server/environment/environment.go

```diff
@@ -86,4 +87,5 @@ type Env interface {
 	//
 	// See server/util/fileresolver/fileresolver.go
 	GetFileResolver() fs.FS
+	GetAdditionalMuxEntries() map[string]http.Handler
```

Comment:
I think instead of putting a map in the env like this you could store a mux in the env and then just add it to the root serve mux in libmain. This is nice because maps are sort of a useful but gross type to pass around in APIs. A mux would be a little bit cleaner.

https://stackoverflow.com/questions/23693520/go-how-to-combine-two-or-more-http-servemux